src/libraries/System.Private.CoreLib/src/System/SearchValues/Strings/Helpers/TeddyHelper.cs

```diff
@@ -0,0 +1,436 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.Arm;
+using System.Runtime.Intrinsics.X86;
+
+namespace System.Buffers
+{
+    /// <summary>
+    /// Contains the implementation of core vectorized Teddy matching operations.
+    /// They determine which buckets contain potential matches for each input position.
+    /// </summary>
+    internal static class TeddyHelper
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [CompExactlyDependsOn(typeof(Ssse3))]
+        [CompExactlyDependsOn(typeof(AdvSimd.Arm64))]
+        public static (Vector128<byte> Result, Vector128<byte> Prev0) ProcessInputN2(
```

Comment:
I've not understood what happens when a match spans across two vectors, e.g. if the first vector ends with ab, the second vector starts with c, and we're looking for abc, how does this find it?

Reply:
Does this comment help?
https://github.com/dotnet/runtime/blob/419a352c8c27c908bf07087970572d198dd72a82/src/libraries/System.Private.CoreLib/src/System/SearchValues/Strings/Helpers/TeddyHelper.cs#L112-L144

I used an example that shows such a scenario ("A" at the end of the first vector, "BC" at the start of the second).